datafusion/functions-aggregate/src/first_last.rs

```diff
@@ -99,19 +130,37 @@ impl AggregateUDFImpl for FirstValue {
         let mut all_sort_orders = vec![];
 
         // Construct PhysicalSortExpr objects from Expr objects:
+        // TODO: Reuse `create_physical_sort_expr`
         let mut sort_exprs = vec![];
         for expr in acc_args.sort_exprs {
-            if let Expr::Sort(sort) = expr {
-                if let Expr::Column(col) = sort.expr.as_ref() {
+            if let Expr::Sort(Sort {
+                expr,
+                asc,
+                nulls_first,
+            }) = expr
+            {
+                if let Expr::Column(col) = expr.as_ref() {
                     let name = &col.name;
                     let e = expressions::column::col(name, acc_args.schema)?;
                     sort_exprs.push(PhysicalSortExpr {
                         expr: e,
                         options: SortOptions {
-                            descending: !sort.asc,
-                            nulls_first: sort.nulls_first,
+                            descending: !asc,
+                            nulls_first: *nulls_first,
                         },
                     });
+                } else if let Expr::Alias(alias) = expr.as_ref() {
```

Comment:
Filed #10074 to see how to arrange the code here, replace with `create_physical_sort_expr` or others